src/gpa.py

```diff
@@ -152,32 +156,50 @@ def compute_results(student_records):
    sessions = get_sessions(student_records)
    semesters = get_semesters(student_records)
    session_total_units, session_total_grade_points =[],[]
-   final_result = []
-   final_result.append(f"Student Name: {student_records['name']}")
-   final_result.append(f"Student ID: {student_records['id']}")
 
+
+   results = list()
+   final_result = {}
+   final_result["Student Name"] = student_records['name']
+   final_result["Student ID"] =  student_records["id"]
+
+   gpa_results ={}
+   gpa_results["Student ID"] =  student_records["id"]
+   
    
    for session in sessions:
+      number_of_sessions = 0
```

Comment:
Why do we need this here when in Ln 156 above we can already determine the number of sessions and semesters?